### PR TITLE
Fix Windows Build

### PR DESCRIPTION
**Pull request by gmittert**
- Set up config and llvm-config.h to not be unix specific
- Import WindowsSupport.h and VerisonTuple.h

Files are copied from swift-llvm c7f5aeb11671f3913b774004484f3f340cbef161, same as the other import.

### CODE REVIEW

**Comment by gmittert** [comment]
@swift-ci please test

File: include/llvm/Config/config.h [hunk 1]
```diff
@@ -51,22 +51,30 @@
...
 #define HAVE_DLADDR 1
+#endif
 
 /* Define to 1 if you have the <errno.h> header file. */
+#if __has_include(<errno.h>)
 #define HAVE_ERRNO_H 1
+#endif
 
 /* Define to 1 if you have the <fcntl.h> header file. */
+#if __has_include(<fnctl.h>)
```
Comment (benlangmuir) [inline include/llvm/Config/config.h]: The "n" and "c" are swapped here, that's what broke the build.

Comment (gmittert) [reply include/llvm/Config/config.h]: Ah, fixed.

**Comment by gmittert** [comment]
@swift-ci please test

**Comment by gmittert** [comment]
@swift-ci please test

**Pull request merged**